internal/core/application/operator_service.go

```diff
@@ -814,6 +827,283 @@ func (o *operatorService) FeeAccountBalance(ctx context.Context) (
 	return int64(baseAssetAmount), nil
 }
 
+type depositType string
+
+var (
+	marketDeposit depositType = "MarketDeposit"
+	feeDeposit    depositType = "FeeDeposit"
+)
+
+func (o *operatorService) ClaimMarketDeposit(
+	ctx context.Context,
+	marketReq Market,
+	outpoints []TxOutpoint,
+) error {
+	addressesPerAccount := make(map[int][]string)
+	bkPairs := make([][]byte, 0)
+
+	err := validateAssetString(marketReq.BaseAsset)
+	if err != nil {
+		return domain.ErrInvalidBaseAsset
+	}
+
+	err = validateAssetString(marketReq.QuoteAsset)
+	if err != nil {
+		return domain.ErrInvalidQuoteAsset
+	}
+
+	if marketReq.BaseAsset != config.GetString(config.BaseAssetKey) {
+		return domain.ErrInvalidBaseAsset
+	}
+
+	market, accountIndex, err := o.marketRepository.GetMarketByAsset(ctx,
+		marketReq.QuoteAsset)
+	if err != nil {
+		return err
+	}
+
+	if market != nil {
+		addresses, bk, err := o.vaultRepository.
+			GetAllDerivedAddressesAndBlindingKeysForAccount(
+				ctx,
+				accountIndex,
+			)
+		if err != nil {
+			return err
+		}
+		addressesPerAccount[accountIndex] = addresses
+		bkPairs = bk
+	} else {
+		vault, err := o.vaultRepository.GetOrCreateVault(
+			ctx,
+			nil,
+			"",
+		)
+		if err != nil {
+			return err
+		}
+		allDerivedAddressInfo := vault.AllDerivedAddressesInfo()
```

Comment:
Would the Vault repository have this method that returns all dervided addresses, already grouped by account index?

Reply:
Explained offline